### ISSUE TITLE
polish endgame is broken

### ISSUE THREAD

**Issue by domino14** [domino14/macondo#154]
```
#character-encoding UTF-8
#player1 1 ptf1559
#player2 2 smut3k
>1: AHIJOUY 8F HUJA +20 20
>1: AHIJOUY --  -20 0
>2: ĆĘIKPST 8G STĘPIĆ +46 46
>1: AHIJOUY 7I HOI +24 24
>2: CFKNWYZ J5 CZ..Y +10 56
>1: AJMSUWY -  +0 24
>2: FKLNŃWW -  +0 56
>1: AJMSUWY -J +0 24
>2: FKLNŃWW -KWŃW +0 56
>1: AAMSUWY 9F AU +14 38
>2: AFLLŁNO 5J .ŁA +12 68
>1: AAMSWYZ 10E SAMY +18 56
>2: CEFLLNO 11C CLE +14 82
>1: AEŁNRWZ 12D AR +13 69
>2: FLNOOOW L3 FL.N +18 100
>1: DEŁNTWZ 13E WENT +10 79
>2: EJNOOOW H13 .EJ +18 118
>1: ?ADIŁZZ 15A ZDZIAŁa. +89 168
>2: IKNOOOW I13 ON +12 130
>1: ABEIMNS 12I SAMBIE +21 189
>2: IIKOOTW M11 K.I +8 138
>1: ?EJNŚWW A14 E. +2 191
>2: AILOOTW C11 .LI +14 152
>1: ?JNNŚWW 12C ...W +7 198
>2: AOOOTWŹ -Ź +0 152
>1: ?AJNNŚW 3L .iŚ +20 218
>2: AGOOOTW K10 GA.O +14 166
>1: AIJNNRW 4L .I +4 222
>2: OORSTWY N12 .WY +10 176
>1: AJNNRRW 11J J. +8 230
>2: ADOORST 12C ....O +8 184
>1: ANNPRRW 9F ..R +10 240
>2: ADORSTY J14 DY +16 200
>1: ACNNPRW 13A CN. +8 248
>2: AOOPRST 10B PO +13 213
>1: ADNPRWZ H8 ...P +7 255
>2: AEKORST 2M KET +25 238
>2: AEKORST --  -25 213
>1: AĄDNRWZ 15J .ARD +7 262
>2: AEKORST 2N ET +18 231
>1: ĄBEGNWZ O1 E. +9 271
>2: AKKORSŹ C10 ....S. +8 239
>1: ĄBGNWZZ 13M ..Ą +7 278
>2: AIKKORŹ M9 RA... +7 246
>1: BGNŃWZZ 10K .N.Ń +26 304
>2: IKKMOŹŻ 9M .OK +12 258
>1: BGHUWZZ N7 ZG.. +12 316
>2: IKMÓŹŻ M1 ŻM.. +13 271
>1: BHUWZ -  +0 316
>2: IKÓŹ B8 KÓ. +9 280
>1: BHUWZ -  +0 316
>2: IŹ 8B .IŹ +21 301
>2:  (BHUWZ) +22 323
```

`set lexicon OSPS44 polish`
`load /path/to/gcg`
`turn 49`
`endgame 10`

load the above gcg and do `turn 49` and `endgame 10` - it generates a bunch of passes in a row. The game should have ended after 2 passes. The more plies are done, the more passes are generated. Wtf?

```
Best sequence has a spread difference of -1
Best sequence:
1) (Pass)
2) (Pass)
3) (Pass)
4) B8 ZU.
5) 9A K.
6) (Pass)
7) 1L I.
8) (Pass)
9) (Pass)
```

**Comment by domino14** [comment]
Correct sequence can be found when we do

`turn 46`
`endgame 20`

```
Best sequence has a spread difference of 5
Best sequence:
1) N7 ZG..
2) M1 ŻM..
3) (Pass)
4) 6L .I
5) B8 ZU.
6) 9A K.
7) (Pass)
8) (Pass)
```

**Comment by domino14** [comment]
fixed in #155 and #157
